packages/frontend/src/Build.svelte

```diff
@@ -0,0 +1,370 @@
+<script lang="ts">
+import './app.css';
+import '@fortawesome/fontawesome-free/css/all.min.css';
+import { faCube, faQuestionCircle, faRocket, faTriangleExclamation } from '@fortawesome/free-solid-svg-icons';
+import { bootcClient } from './api/client';
+import FormPage from './lib/upstream/FormPage.svelte';
+import Input from './lib/upstream/Input.svelte';
+import Button from './lib/upstream/Button.svelte';
+import type { BootcBuildInfo } from '@shared/src/models/bootc';
+import Fa from 'svelte-fa';
+import { onMount } from 'svelte';
+import type { ImageInfo } from '@podman-desktop/api';
+import EmptyScreen from './lib/upstream/EmptyScreen.svelte';
+import { router } from 'tinro';
+
+// Image variables
+let selectedImage: ImageInfo;
+let buildImageName: string;
+let buildTag: string;
+let buildEngineId: string;
+
+// Build options
+let buildFolder: string;
+let buildType: string;
+let buildArch: string;
+
+// Other variables
+let success = false;
+let buildInProgress = false;
+let bootcAvailableImages: any[] = [];
+let errorMessage = '';
+let errorFormValidation = '';
+
+// Function that will use listHistoryInfo, if there is anything in the list, pick the first one in the list (as it's the most recent)
+// and fill buildFolder, buildType and buildArch with the values from the selected image.
+async function fillBuildOptions() {
+  const historyInfo = await bootcClient.listHistoryInfo();
+  if (historyInfo.length > 0) {
+    const latestBuild = historyInfo[0];
+    buildFolder = latestBuild.folder;
+    buildType = latestBuild.type;
+    buildArch = latestBuild.arch;
+
+    // Go through bootcAvailableImages and find the one that matches latestBuild.image and latestBuild.tag
+    // and set selectedImage to that value.
+    for (const image of bootcAvailableImages) {
+      if (image.RepoTags && image.RepoTags.length > 0) {
+        if (image.RepoTags[0] === `${latestBuild.name}:${latestBuild.tag}`) {
+          selectedImage = image;
+          break;
+        }
+      }
+    }
+  }
+}
+
+async function buildBootcImage() {
+  // Before ANYTHING validate / check all values before submitting:
+  // Reset the form validation error message
+  errorFormValidation = '';
+  let missingFields = [];
+  if (!buildImageName) missingFields.push('image name');
+  if (!buildTag) missingFields.push('tag');
+  if (!buildEngineId) missingFields.push('engine ID');
+  if (!buildFolder) missingFields.push('build folder');
+  if (!buildType) missingFields.push('build type');
+  if (!buildArch) missingFields.push('architecture');
```

Comment:
I think we're requiring brackets
if (!buildImageName) {
  missingFields.push('image name');
}

Reply:
or I don't know if you could `buildImageName ?? missingFields.push('image name')`